### ISSUE TITLE
PC/MacOS(?)

### ISSUE THREAD

**Issue by khaosn** [ptarroso/speciesdigitizer#1]
Hi, just found your site, looks very promising for what I was trying to do. Installation worked fine but after the dialogue box is opened and the image file and ControlPoints are located it doesn't go further (i.e. actually show the image). Is it a MacOS/PC thing or something else?

**Comment by ptarroso** [comment]
Hi

I just tried on linux and Windows (latest R and Rstudio versions) and works ok. Don't have access to any Mac now... Do you have any error that you can provide here? 

I believe this package does not work so well with Rstudio as it need some degree of interaction with the plot. Maybe you can try to bypass RStudio. Can you try it in the console that comes with R (RGui in windows, not sure in Mac) to see if the problem persists?

Other thing you might try is to use directly the R console. On Windows you can find the "cmd" program in the start menu and then type "c:\Program Files\R\R-4.2.2\bin\R" (with the quotes) and try it there. On Mac open the terminal and start by typing "R" in the command line (without quotes).

**Comment by khaosn** [comment]
Hi, thanks for the reply. I tried it on the built-in R editor and it worked
much better, thanks for the advice. I had a few glitches: after hitting
>Final check<, I tried to use the + and - buttons, but got no reaction and
the windows kind of stalled. I also could not save the points when hitting
<Write file> so I manually closed the window, at which point the dialogue
window for writing file appeared (and I could save the points). I had the
georeferenced map ticked, but I got an error message saying that
GeoTransform values are not available.

Error in locator() :
  graphics device closed during call to locator or identify
In addition: Warning message:
In readGDAL(fileName, silent = TRUE) : GeoTransform values not available
Error in locator() : plot.new has not been called yet
Error in locator() : plot.new has not been called yet

Repeated the attempt a few times.

R on Mac frequently makes use of XQuartz/X11 app (https://www.xquartz.org)
for interactive windows (not always, but it does here) - perhaps this is
interfering with the aspects of the functionality?

Nevertheless, I was able to do what I wanted with my old map images - so
many thanks for this really useful package! Out of curiosity, will you
update it to be functional after rgdal is retired?

/Karin

**Comment by ptarroso** [comment]
Hi, I'm glad you manage to digitize your maps with this tool! 

Sometimes the final check can be slow, especially if you have a very large grid with lots of points identified. 
I made this script to help my own work and later turn it into a package as it can be helpful to others. But I did not put much effort on solving some glitches or being more efficient as it solved my own problems as it is. Thus, there might be some issues, particularly on the interface that can be improved.

The locator() is a native R function that requires an open graphic device (locates coordinates of a mouse click in the plot). If you close the device, than the locator gives that error. 

I'm not sure also how much the xquartz interferes with the package... The package is not particularly fast with very large images, so I would use the minimum resolution possible when you still able to identify the points in the map. So, if your input image shows just a few pixels per map point to be identified, it should be ok to use with the package. On the other hand, if you have a very large resolution (large amount of pixels per map point) than it will be slower and increase the number of problems in the digitization. 

Regarding the rgdal end of life, I will try to switch to other packages (likely terra) but I cannot make any promise when it will happen! :)

Thanks for the feedback!
I will make this thread as solved!

all the best!
